gcp/gce_instance_test.go

```diff
@@ -0,0 +1,128 @@
+	"strings"
+	"testing"
+	"time"
+)
+
+func testZone() string {
+	return "us-central1-c"
+}
+
+func resourcesContains(resources []GcpResource, name string) bool {
+	for _, resource := range resources {
+		if resource.Name() == name && resource.Location() == testZone() {
+			return true
+		}
+	}
+
+	return false
+}
+
+func createTestInstance(ctx *GcpContext, name string, protected bool) error {
+	machineType := fmt.Sprintf("projects/%s/zones/%s/machineTypes/f1-micro", ctx.Project, testZone())
+
+	iface := &compute.NetworkInterface{}
+
+	initializeParams := &compute.AttachedDiskInitializeParams{
+		SourceImage: "projects/debian-cloud/global/images/debian-9-stretch-v20181113",
+	}
+
+	disk := &compute.AttachedDisk{
+		AutoDelete:       true,
+		Boot:             true,
+		InitializeParams: initializeParams,
+	}
+
+	instance := &compute.Instance{
+		Name:               name,
+		MachineType:        machineType,
+		DeletionProtection: protected,
+		NetworkInterfaces:  []*compute.NetworkInterface{iface},
+		Disks:              []*compute.AttachedDisk{disk},
+	}
+
+	_, err := ctx.Service.Instances.Insert(ctx.Project, testZone(), instance).Do()
+	return err
+}
+
+func TestNukeInstances(t *testing.T) {
+	t.Parallel()
+
+	ctx, err := DefaultContext()
+	if err != nil {
+		assert.Fail(t, gruntworkerrors.WithStackTrace(err).Error())
+		return
+	}
+
+	instanceName := strings.ToLower("cloud-nuke-test-" + util.UniqueID())
+	protectedInstanceName := strings.ToLower("cloud-nuke-test-" + util.UniqueID())
+
+	defer func() {
+		call := ctx.Service.Instances.SetDeletionProtection(ctx.Project, testZone(), protectedInstanceName)
+		call.DeletionProtection(false)
+
+		_, err := call.Do()
+		if err != nil {
+			assert.Fail(t, gruntworkerrors.WithStackTrace(err).Error())
+		}
+
+		_, err = ctx.Service.Instances.Delete(ctx.Project, testZone(), protectedInstanceName).Do()
+		if err != nil {
+			assert.Fail(t, gruntworkerrors.WithStackTrace(err).Error())
+		}
+	}()
+
+	err = createTestInstance(ctx, instanceName, false)
+	if err != nil {
+		assert.Fail(t, gruntworkerrors.WithStackTrace(err).Error())
+	}
+
+	err = createTestInstance(ctx, protectedInstanceName, true)
+	if err != nil {
+		assert.Fail(t, gruntworkerrors.WithStackTrace(err).Error())
+	}
+
+	instances, err := GetAllGceInstances(ctx, []string{}, time.Now().Add(1*time.Hour))
+	if err != nil {
+		assert.Fail(t, gruntworkerrors.WithStackTrace(err).Error())
+	}
+
+	assert.True(t, resourcesContains(instances, instanceName),
+		"the created instance should show up in the list of instances")
+	assert.False(t, resourcesContains(instances, protectedInstanceName),
+		"the protected instance should not show up in the list of instances")
+
+	err = ctx.NukeAllResources(instances)
+	if err != nil {
+		assert.Fail(t, gruntworkerrors.WithStackTrace(err).Error())
+	}
+
+	// status doesn't update immediately, so give it a minute to show it is
+	// terminating
+	tries := 0
+	lastStatus := ""
+	for tries < 20 {
```

Comment:
Looks like you're not incrementing `tries`? Use `for tries := 0; tries < 20; tries++`.